### PR TITLE
Use `<Spoiler>` component directly & remove micromark extension

### PR DESCRIPTION
**Pull request by delucis**
#### What kind of changes does this PR include?

- Changes to the docs site code

#### Description

- Replaces our custom `||spoiler||` syntax with standard `<Spoiler>spoiler</Spoiler>` component usage in the tutorial, fixing rendering bugs with our micromark extension since the MDX migration.
- Removes all the code for the spoiler micromark extension & component auto-imports now we’re no longer using them.

Here’s an example page this PR fixes:

- **Before:** https://63b826bed46e4f0009cce151--astro-docs-2.netlify.app/en/tutorial/6-islands/1/#analyze-the-pattern
- **After:** https://deploy-preview-2308--astro-docs-2.netlify.app/en/tutorial/6-islands/1/#analyze-the-pattern

### CODE REVIEW

File: src/pages/en/tutorial/2-pages/5.mdx [hunk 1]
```diff
@@ -108,19 +109,27 @@ Your About page is now styled using *both* the imported `global.css` file *and*
...
-    || Yes ||
+    <p>
+      <Spoiler>Yes</Spoiler>
+    </p>
 
 - Are there any conflicting styles, and if so, which are applied?
 
-    || Yes, h1 has a size of 2.5rem globally, but 4rem locally in the <style> tag. The local 4rem rule is applied on the About page. ||
+    <p>
+      <Spoiler>Yes, h1 has a size of 2.5rem globally, but 4rem locally in the `<style>` tag. The local 4rem rule is applied on the About page.</Spoiler>
```
Comment (sarah11918) [inline src/pages/en/tutorial/2-pages/5.mdx:119]:
```suggestion
      <Spoiler>Yes, `<h1>`  has a size of `2.5rem` globally, but `4rem` locally in the `<style>` tag. The local `4rem` rule is applied on the About page.</Spoiler>
```
FWIW, this is what we had before. I think making h1 a tag makes sense, and open to offers on whether we want the `rem` units back ticked!

File: src/pages/en/tutorial/2-pages/5.mdx [hunk 2]
```diff
@@ -108,19 +109,27 @@ Your About page is now styled using *both* the imported `global.css` file *and*
...
+    <p>
+      <Spoiler>When conflicting styles are defined both globally and in a page's local `<style>` tag, the local styles should overwrite any global styles. (But, there can be other factors involved, so always visually inspect your site to make sure your styles are properly applied!)</Spoiler>
+    </p>
 
 - How would you choose whether to declare a style in a `global.css` file or a `<style>` tag?
 
-    || If you want a style to be applied site-wide, you would choose to use a global.css file. However, if you want styles to apply to only the HTML content in a single Astro file, and not affect other elements on your site, you would choose a <style> tag. ||
+    <p>
+      <Spoiler>If you want a style to be applied site-wide, you would choose to use a global.css file. However, if you want styles to apply to only the HTML content in a single Astro file, and not affect other elements on your site, you would choose a `<style>` tag.</Spoiler>
```
Comment (sarah11918) [inline src/pages/en/tutorial/2-pages/5.mdx:131]:
```suggestion
      <Spoiler>If you want a style to be applied site-wide, you would choose to use a `global.css` file. However, if you want styles to apply to only the HTML content in a single `.astro` file, and not affect other elements on your site, you would choose a `<style>` tag.</Spoiler>
```
Again, what we had before, if we want!

**Review by sarah11918 — APPROVED** [review]
They all work for me now! Just two places where we might want to reintroduce some code styling.

File: src/pages/en/tutorial/2-pages/5.mdx (hunk 1 above)
Comment (delucis) [reply src/pages/en/tutorial/2-pages/5.mdx:119]: Oh yes, should definitely add these back 👍

**Pull request merged**